src/core/SampleBuffer.cpp

```diff
@@ -1017,30 +1017,56 @@ void SampleBuffer::visualize(
 	const float ySpace = h * 0.5f;
 	const int nbFrames = focusOnRange ? toFrame - fromFrame : m_frames;
 
-	const int fpp = qBound<int>(1, nbFrames / w, 20);
-	QPointF * l = new QPointF[nbFrames / fpp + 1];
-	QPointF * r = new QPointF[nbFrames / fpp + 1];
-	int n = 0;
+	const int fpp = qMax<int>(1, nbFrames / w);
+	const int totalPoints = nbFrames / fpp * 2 + 2;
```

Comment:
```suggestion
	const int totalPoints = nbFrames / fpp * 2;
```

After reading the method, I still can't see why those extra 2 points are necessary. It seems it has been tested without it with no issues. Just for safety maybe you could remove it and add an statement on the `for` loop, something like:
`if((curFrame + 1) >= totalPoints) { qWarning("Out of bounds"); }`

Just temporarily, and test it with lots of samples to see whether any of them would trigger it. But from reading the logic it doesn't seem to me that it would ever reach that index.

Reply:
I just tested it out without the +2 and the program would abort entirely every time I tried doing anything with a sample track.  I assume it's needed for some reason.

Reply:
That's odd. I tried to use the Peaks/Troughs approach on that Oscillator plugin but I used a slightly different approach, so it's hard to compare but I didn't need extra points. Did you manage to get a backtrace on the crashes?

I'm trying to debug it here to figure it out, I'm just a bit averse to leave it there without at least knowing what it's for, even though it was already there :laughing: . If it's necessary, I'll try to figure out why so we can at least add a comment.

Reply:
Small test indicates the logic is has a small flaw: We are drawing past the width of the widget:

```
nbFrames:  166623
Width:  35
fpp:  4760
totalPoints:  72
first:  0
last:  166622

Out of bounds
curFrame:  35
index:  71
x:  35.9953


Out of bounds
curFrame:  36
index:  73
x:  36.9952
```

As can be seen, both "out of bound access" (that reach those extra two points) have an X value higher then the widget's width. 166623 frames is not perfectly divisible by 35 pixels (results in 4760.66), so the `fpp` is floored to 4760. By fitting the 35 pixels we have consumed `4760 * 35` frames, which is 166600. Those extra 23 frames are being drawn too when they should be discarded apparently. If not discarded, they should be used only for the peak/trough calculation of pixel 35.

I'll later try to come up with some fix, but at least the mystery seems to be solved :smile:

Reply:
Also, somewhat unrelated, but can `curFrame` be renamed to `curPixel`? Think it's a more accurate name